stdlib/public/core/Optional.swift

```diff
@@ -487,3 +487,78 @@ extension Optional {
   }
 
 }
+
+//===----------------------------------------------------------------------===//
+// Bridging
+//===----------------------------------------------------------------------===//
+
+#if _runtime(_ObjC)
+extension Optional : _ObjectiveCBridgeable {
+  // The object that represents `none` for an Optional of this type.
+  internal static var _nilSentinel : AnyObject {
+    @_silgen_name("_swift_Foundation_getOptionalNilSentinelObject")
+    get
+  }
+
+  public func _bridgeToObjectiveC() -> AnyObject {
+    // Bridge a wrapped value by unwrapping.
+    if let value = self {
+      return _bridgeAnythingToObjectiveC(value)
+    }
+    // Bridge nil using a sentinel.
+    return type(of: self)._nilSentinel
+  }
+
+  public static func _forceBridgeFromObjectiveC(_ source: AnyObject,
+                                         result: inout Optional<Wrapped>?) {
+    // Map the nil sentinel back to .none.
+    // NB that the signature of _forceBridgeFromObjectiveC adds another level
+    // of optionality, so we need to wrap the immediate result of the conversion
+    // in `.some`.
+    if source === _nilSentinel {
```

Comment:
It is documented as a singleton, but I would double-check with someone from Foundation. @phausler?